File: app/src/main/java/com/waygo/data/model/fuel/Electric.java
package com.waygo.data.model.fuel;


import android.support.annotation.NonNull;

import com.waygo.utilskt.Result;


public final class Electric extends Fuel {
    protected Electric(float value) {
        super(value);
    }

    @NonNull
    public static Result<Fuel> create(final float value) {
        return createValue(value).map(v -> (Fuel) new Electric(v));
    }
}

File: app/src/main/java/com/waygo/data/model/fuel/Electric.kt
package com.waygo.data.model.fuel

import com.waygo.utilskt.Result
import com.waygo.utilskt.ofType
import kotlin.platform.platformStatic

public class Electric protected constructor(value: Float) : Fuel(value) {
    companion object {
        public platformStatic fun create(value: Float): Result<Fuel> =
                Fuel.createValue(value)
                    .map({Electric})
                    .ofType<Fuel>()
    }
}
